src/middlewared/middlewared/plugins/apps/ix_apps/query.py

```diff
@@ -47,6 +47,16 @@ def normalize_portal_uri(portal_uri: str, host_ip: str | None) -> str:
     if not host_ip or '0.0.0.0' not in portal_uri:
         return portal_uri
 
+    # 'portal_uri' contains '0.0.0.0:<port>'.
+    # We want to replace the '0.0.0.0' part with the IP address of the host.
+    # However, if the host IP address is an IPv6 address,
+    # we need to wrap it in brackets '[]' in order to produce a valid URI.
+    # We already assume that host_ip doesn't contain a port number,
+    # so checking if it contains a ':' should be sufficient to determine
+    # whether or not it's an IPv6 address.
+    if ':' in host_ip:
```

Comment:
@mortie we have added changes where apps now will already have the portal URI normalized if they are ipv6 based (https://github.com/truenas/middleware/blob/4f404a59ec31c53606a8352ad209c38954610886/src/middlewared/middlewared/plugins/apps/ix_apps/portals.py#L12).

However people who are using apps which were installed/modified before they come to this version, they will run into this issue. Based off this, we definitely would not want to have this normalized 2 times so we should check that `[` is not already present and `:` is present and then normalize it